tubular/edx_api.py

```
@@ -29,45 +29,70 @@ class BaseApiClient:
     API client base class used to submit API requests to a particular web service.
     """
     append_slash = True
-    _client = None
+    _access_token = None
 
     def __init__(self, lms_base_url, api_base_url, client_id, client_secret):
         """
         Retrieves OAuth access token from the LMS and creates REST API client instance.
         """
         self.api_base_url = api_base_url
-        access_token, __ = self.get_access_token(lms_base_url, client_id, client_secret)
-        self.create_client(access_token)
+        self._access_token = self.get_access_token(lms_base_url, client_id, client_secret)
 
-    def create_client(self, access_token):
+    def get_api_url(self, path):
         """
-        Creates and stores the EdxRestApiClient that we use to actually make requests.
+        Construct the full API URL using the api_base_url and path.
+        Args:
+            path (str): API endpoint path.
         """
-        self._client = EdxRestApiClient(
-            self.api_base_url,
-            jwt=access_token,
-            append_slash=self.append_slash
-        )
+        path = path.strip('/')
+        if self.append_slash:
+            path += '/'
+
+        return urljoin(f"{self.api_base_url}/", path)
+
+    def _request(self, method, url, **kwargs):
+        response = requests.request(method, url, auth=SuppliedJwtAuth(self._access_token), **kwargs)
+
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            exc.content = response.content
+            raise exc
```

If we need to reraise the same exception it should be just raise